startups/lib/main_9.dart

```diff
@@ -0,0 +1,220 @@
+import 'package:english_words/english_words.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (BuildContext context) => MyAppState(),
+      child: MaterialApp(
+        title: 'Namer App',
+        theme: ThemeData(
+          useMaterial3: true,
+          colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepOrange),
+        ),
+        home: MyHomePage(),
+      ),
+    );
+  }
+}
+
+class MyAppState extends ChangeNotifier {
+  var current = WordPair.random();
+
+  var history = <WordPair>[];
+
+  var favorites = <WordPair>[];
```

Comment:
I was really hoping to have just `var history = [];` here, but the type information is really helpful when writing consumer code.

Reply:
Does history the var (not the list) ever get modified? Making this lists final and modifying them in place makes a lot of sense, and offers a way to subtly hint that Dart isn't JS

Reply:
As above, I see this as something to _tell_ to devs, but not requiring the beginner students to understand it or work with it. After all, the app works just fine with `var`, and at this level, `final` doesn't give much. But you're right that the more experienced devs could read this as Dart being "weak". Better to make sure they understand there's more.